src/relay/op/nn/nn.cc

```diff
@@ -1177,7 +1177,8 @@ bool NLLLossRel(const Array<Type>& types, int num_inputs, const Attrs& attrs,
                                      << ", weights shape = " << weights->shape);
     return false;
   }
-  if (!(predictions->dtype == weights->dtype && predictions->dtype.is_float())) {
+  if (!(predictions->dtype == weights->dtype &&
+        (predictions->dtype.is_float() || predictions->dtype.is_bfloat16()))) {
```

Comment:
Shall we let `is_float()` be true for `bfloat16` exprs?

Reply:
I prefer this way too, since they **are** all floating-point datatypes.
While there are some practice inconsistences so far, for example, if we let `is_float() == true` for `bfloat16`, then we cannot distinguish `bfloat16` and `float16` anymore as they both satisfy the condition `is_float() == true && bits() == 16`.